airflow/www/views.py

```diff
@@ -2298,8 +2298,10 @@ def duration(self, session=None):
         else:
             base_date = dag.get_latest_execution_date() or timezone.utcnow()
 
-        dates = dag.date_range(base_date, num=-abs(num_runs))
-        min_date = dates[0] if dates else timezone.utc_epoch()
+        min_date = next(
+            dag.time_table.iter_next_n(base_date, 1),
+            timezone.utc_epoch(),
+        )
```

Comment:
This was powering the "dag runs" drop down on the graph view etc.

![image](https://user-images.githubusercontent.com/34150/120627600-bd2f6500-c45b-11eb-9399-c1d8adb15b49.png)

So this change will break the values in the drop down (and yes, it's not covered by tests. UI tests are hard/bad)

Reply:
1. Don’t do `abs()` in `iter_next_n()`, raise a `ValueError`.
2. Change the places that are relying on `iter_next_n()` to return previous runs to get the actual `DagRun`s instead.

Reply:
Oh no, it's worse that that.

This controls what runs are shown on the these views, where "base_date" is current and then we look back `num_runs` older runs.

Reply:
I think this needs to be something like:

```
SELECT execution_date FROM dag_run WHERE .... ORDER BY execution_date DESC OFFSET ? LIMIT 1
```

with `abs(num_runs)` as the offset.

I _think_

Reply:
Possibly also adding in a "type != manual" too.